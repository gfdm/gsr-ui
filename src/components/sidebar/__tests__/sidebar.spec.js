import { mount, createLocalVue } from '@vue/test-utils'
import IdSidebar from '../'

const localVue = createLocalVue()
localVue.use(IdSidebar)

describe('Sidebar', () => {
  test('should render well', () => {
    const Wrapper = {
      name: 'wrap',
      template: `
        <id-sidebar>
          <id-sidebar-group title="Title">
            <id-sidebar-item>A</id-sidebar-item>
            <id-sidebar-item>B</id-sidebar-item>
          </id-sidebar-group>
          <id-sidebar-item>C</id-sidebar-item>
        </id-sidebar>
      `
    }

    const wrapper = mount(Wrapper, {
      localVue
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
