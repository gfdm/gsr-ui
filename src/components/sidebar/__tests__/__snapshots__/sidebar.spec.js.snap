// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should render well 1`] = `
<aside class="id-sidebar">
  <header class="id-sidebar__header"></header>
  <ul>
    <li class="id-sidebar-group">
      <div class="id-sidebar-group__title">
        Title
      </div>
      <ul>
        <li class="id-sidebar-item"><span class="id-sidebar-item__text">A</span></li>
        <li class="id-sidebar-item"><span class="id-sidebar-item__text">B</span></li>
      </ul>
    </li>
    <li class="id-sidebar-item"><span class="id-sidebar-item__text">C</span></li>
  </ul>
  <footer class="id-sidebar__footer"></footer>
</aside>
`;
